Data/EmulatedDevice.cs

```diff
@@ -221,14 +221,57 @@ public ICommand AddKeyCommand
             }
         }
 
+        [JsonIgnore]
+        public ICommand AddAllKeyCommand
+        {
+            get { return new RelayCommand(AddAllKey); }
+        }
+
+        private void AddAllKey(object obj)
+        {
+            string name;
+            while ((name = nextname()) != "")
+            {
+                AddKey(name);
+            }
+        }
+
         private void AddKey(object obj)
         {
+            string name;
+            name = obj == null? nextname() : obj.ToString();
+
             EmulatedKeys.Add(
                 new EmulatedKey
                 {
-                    Name = ""
+                    Name = name
                 });
         }
+
+        /// <summary>
+        /// Search for a key name that is not in use.
+        /// </summary>
+        private string nextname()
```

Comment:
Please use camelcase here.  Also maybe `GetNextSuggestion` would be more informative?